### PR TITLE
feat: New Module `avm/res/healthcare-apis/workspace`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/healthcare-apis/workspace`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.healthcare-apis.workspace](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.healthcare-apis.workspace.yml) |

### CODE REVIEW

File: avm/res/healthcare-apis/workspace/dicomservice/main.bicep [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+@description('The resource group where the namespace is deployed.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(
+    dicom.identity,
+    'principalId'
+  )
+  ? dicom.identity.principalId
+  : ''
```
Comment (AlexanderSehr) [inline avm/res/healthcare-apis/workspace/dicomservice/main.bicep:154]:
```suggestion
@description('The principal ID of the system assigned identity.')
output systemAssignedMIPrincipalId string = dicom.?identity.?principalId ?? ''
```

File: avm/res/healthcare-apis/workspace/fhirservice/main.bicep [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+    }
+  : null
+
+var accessPolicies = [
+  for id in accessPolicyObjectIds ?? []: {
+    objectId: id
+  }
+]
+
+var exportConfiguration = {
```
Comment (AlexanderSehr) [inline avm/res/healthcare-apis/workspace/fhirservice/main.bicep:130]: The var is only used in a single location. Should not be a var

File: avm/res/healthcare-apis/workspace/iotconnector/main.bicep [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+@description('The resource group where the namespace is deployed.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(
+    iotConnector.identity,
+    'principalId'
+  )
+  ? iotConnector.identity.principalId
+  : ''
```
Comment (AlexanderSehr) [inline avm/res/healthcare-apis/workspace/iotconnector/main.bicep:156]:
```suggestion
output systemAssignedMIPrincipalId string =  iotConnector.?identity.?principalId ?? ''
```

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]

File: avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    publicNetworkAccess: 'Enabled'
```
Comment (AlexanderSehr) [inline avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep:43]: Required?

File: avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        diagnosticSettings: [
+          {
+            eventHubName: diagnosticDependencies.outputs.eventHubNamespaceEventHubName
+            eventHubAuthorizationRuleResourceId: diagnosticDependencies.outputs.eventHubAuthorizationRuleId
+            storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
+            workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
+          }
+        ]
+        publicNetworkAccess: 'Enabled'
+        managedIdentities: {
```
Comment (AlexanderSehr) [inline avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep:118]: Are all these parameters required for WAF-alignment?

File: avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
```
Comment (AlexanderSehr) [inline avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep:33]: needed?

File: avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
```
Comment (eriqua) [inline avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep:37]: Is the idempotency test skipped intentionally?

Comment (krbar) [reply avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep:37]: > Is the idempotency test skipped intentionally?

No, let me check it, thanks for pointing out.

File: avm/res/healthcare-apis/workspace/fhirservice/main.bicep (hunk 2 above)
Comment (krbar) [reply avm/res/healthcare-apis/workspace/fhirservice/main.bicep:124]: Removed the variable: https://github.com/Azure/bicep-registry-modules/pull/1467/commits/277fb756d1b61515cc9afafbd3b6a13315ea75fe

File: avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep (hunk 4 above)
Comment (krbar) [reply avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep]: Removed the non-required param

File: avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep (hunk 6 above)
Comment (krbar) [reply avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep]: removed

File: avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep (hunk 5 above)
Comment (krbar) [reply avm/res/healthcare-apis/workspace/tests/e2e/waf-aligned/main.test.bicep]: removed the child module tests

File: avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep (hunk 7 above)
Comment (krbar) [reply avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep]: Added the idempotency tests, the tests were successful, no idea why the CARML module didn't have them...

**Review by AlexanderSehr — APPROVED** [review]

File: avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep (hunk 7 above)
Comment (AlexanderSehr) [reply avm/res/healthcare-apis/workspace/tests/e2e/defaults/main.test.bicep]: checked & resolved

**Pull request merged**